README.md

```diff
@@ -1,38 +1,81 @@
 
 
 ![alt text](figures/figure-2-color.jpeg)
 
-## Dependencies
-- [Python 3.6](https://www.anaconda.com/download/#macos)
-- [PyTorch](https://pytorch.org/)
-- [PIL](https://pillow.readthedocs.io/en/5.3.x/)
+## Changes in this Release
+- Use Python [multiprocessing](https://docs.python.org/3.6/library/multiprocessing.html)
+to speed up patch creation.
+- Replace [os](https://docs.python.org/3.6/library/os.html) with
+[pathlib](https://docs.python.org/3.6/library/pathlib.html) module.
+- Added [argparse](https://docs.python.org/3.6/library/argparse.html) support.
+- Rewrote patch processing code using vector operations.
+- Fully documented all functions and methods with docstrings and type hints.
+- Compute image means and standard deviations for normalization online.
+- Increased efficiency of code by rewriting inefficient code.
+
+## Known Issues and Limitations
+- Patch creation code doesn't find any patches in images that are slightly larger
+than the specified patch size.
+- Only 1 GPU supported.
+- Should work, but not tested on Windows.
+- In cases where no crops are found for an image, empty directories are created.
+Current workaround uses `try` and `except` statements to catch errors.
+- Image reading code expects colors to be in the RGB space. Current workaround
+is to keep first 3 channels.
+
+## Requirements
+- [imageio](https://pypi.org/project/imageio/)
+- [NumPy 1.16](https://numpy.org/)
 - [OpenCV](https://opencv.org/)
-- [Scikit-Learn](https://scikit-learn.org/stable/install.html)
-- [Scipy](https://www.scipy.org/)
-- GPU
+- [OpenSlide](https://openslide.org/)
+- [OpenSlide Python](https://openslide.org/api/python/)
+- [pandas](https://pandas.pydata.org/)
+- [PIL](https://pillow.readthedocs.io/en/5.3.x/)
+- [Python 3.6](https://www.python.org/downloads/release/python-360/)
+- [PyTorch](https://pytorch.org/)
+- [scikit-image](https://scikit-image.org/)
+- [scikit-learn](https://scikit-learn.org/stable/install.html)
+- [SciPy](https://www.scipy.org/)
+- [NVIDIA GPU](https://www.nvidia.com/en-us/)
+- [Ubuntu](https://ubuntu.com/)
 
 # Usage
 
 Take a look at `code/config.py` before you begin to get a feel for what parameters can be changed.
 
 ## 1. Train-Val-Test Split:
 
-Splits the data into a validation and test set. Default validation whole-slide images (WSI) per class is 15 and test images per class is 30. You can change these numbers by changing `val_wsi_per_class` and `test_wsi_per_class` in `code/config.py`. You can skip this step if you did a custom split (for example, you need to split by patients).
+Splits the data into a validation and test set. 
+Default validation whole-slide images (WSI) per class is 20 and test images per class is 30. 
+You can change these numbers by changing the `--val-wsi-per-class` and \
```

Comment:
Should these be in snake case? I.e., --val_wsi_per_class

Reply:
@jasonwei20 The flags are currently expected to be in the `--val-wsi-per-class` form.

Reply:
See my other comment

Reply:
@jasonwei20 I'll switch the config file to snake case.